### PR TITLE
ci/Add code coverage

### PR DESCRIPTION
**Pull request by Magnus-Kuhn**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -31,6 +31,11 @@ jobs:
           NMSHD_TEST_BASEURL: http://localhost:8090
           NMSHD_TEST_CLIENTID: test
           NMSHD_TEST_CLIENTSECRET: test
+      - name: Upload coverage reports to Codecov with GitHub Action
+        uses: codecov/codecov-action@v4
+        env:
+          fail_ci_if_error: true
```
Comment (jkoenig134) [inline .github/workflows/test.yml:37]: I'm not a fan of this (in fact we should also disable this in the connector again). coverage should only be an information for us and not a hard requirement.

File: .github/workflows/test.yml
Comment (jkoenig134) [inline .github/workflows/test.yml:1]: the triggers for this workflow should be moved from 

```
on:
  push:
    branches-ignore: [main]
```

to 

```
on:
  pull_request:
  push:
    branches: [main]
```

File: packages/app-runtime/package.json [hunk 2]
```diff
@@ -23,11 +23,21 @@
         "cdep": "tsc && madge --circular dist",
         "lint:tsc": "tsc --noEmit && tsc -p test/tsconfig.json --noEmit",
         "test": "jest --forceExit",
-        "test:ci": "jest -i --forceExit"
+        "test:coverage": "npm run test -- --coverage=true",
+        "test:ci": "jest -i --forceExit --coverage=true"
```
Comment (jkoenig134) [inline packages/app-runtime/package.json:27]:
```suggestion
        "test:coverage": "npm run test -- --coverage",
        "test:ci": "jest -i --forceExit --coverage"
```

same for all other occurences.

File: packages/consumption/package.json [hunk 3]
```diff
@@ -24,17 +24,27 @@
         "lint:tsc": "tsc --noEmit && tsc -p test/tsconfig.json --noEmit",
         "test:ci": "npm run test",
         "test:ci:lokijs": "USE_LOKIJS=true jest -i",
-        "test:ci:mongodb": "jest -i",
+        "test:ci:mongodb": "jest -i  --coverage=true",
         "test:local:ferretdb": "npm run test:local:start:ferretdb && CONNECTION_STRING='mongodb://localhost:27022' jest",
         "test:local:lokijs": "USE_LOKIJS=true jest",
         "test:local:mongodb": "npm run test:local:start:mongodb && CONNECTION_STRING='mongodb://root:example@localhost:27021' jest",
+        "test:local:coverage": "npm run test:local:lokijs -- --coverage=true",
```
Comment (jkoenig134) [inline packages/consumption/package.json:31]: IMO we don't need the local scripts. What if you want to test the coverage against mongodb? just run `npm run test:local:mongodb -- --coverage` same for **all** other places.

File: packages/runtime/package.json [hunk 4]
```diff
@@ -24,17 +24,27 @@
         "cdep": "tsc && madge --circular dist",
         "lint:tsc": "tsc --noEmit && tsc -p test/tsconfig.json --noEmit",
         "test:ci:lokijs": "USE_LOKIJS=true jest -i",
-        "test:ci:mongodb": "jest -i",
+        "test:ci:mongodb": "jest -i  --coverage=true",
```
Comment (jkoenig134) [inline packages/runtime/package.json:27]:
```suggestion
        "test:ci:mongodb": "jest -i --coverage=true",
```

same for all other places.

**Comment by jkoenig134** [comment]
some details. Overall looks good.

@sebbi08 please also give your sense as you made the changes on the connector.

File: .github/workflows/test.yml (hunk 1 above)
Comment (sebbi08) [reply .github/workflows/test.yml:37]: That option has nothing to do with the coverage check, it is only for the upload of the coverage files.
We can disable this, but I think the CI check needs to be disabled in the codecov config.

Comment (jkoenig134) [reply .github/workflows/test.yml:37]: Ah, IC. then nvm. Upload errors SHOULD fail this ppl.

File: packages/consumption/package.json (hunk 3 above)
Comment (sebbi08) [reply packages/consumption/package.json:31]: I don't like it either that there are so many npm commands.

I also think we should remove it and maybe document it in the dev README

Comment (Magnus-Kuhn) [reply packages/consumption/package.json:31]: > What if you want to test the coverage against mongodb?

Does this matter?
I can change it, I simply copied it from the connector ;)

Comment (jkoenig134) [reply packages/consumption/package.json]: yeah, it does.

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**